tests/unit/htdocs/modules/protector/ProtectorMysqlDatabaseTest.php

```
@@ -224,4 +224,93 @@ public function doubtfulRequestsCanBeSet(): void
+    {
+        $ref = new \ReflectionMethod(\ProtectorMySQLDatabase::class, 'exec');
+        $returnType = $ref->getReturnType();
+        $this->assertNotNull($returnType);
+        $this->assertSame('bool', $returnType->getName());
+    }
+
+    #[Test]
+    public function execCallsCheckSqlOnDoubtfulNeedle(): void
+    {
+        // Create a partial mock to verify checkSql is called
+        $ref = new \ReflectionClass(\ProtectorMySQLDatabase::class);
```

PHPMD (2.15.0)

[error] Missing class import via use statement (line '272', column '20'). (undefined)

(MissingImport)